client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditModal.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import {
+  DialogButton,
+  ModalMode,
+  useBufferState,
+  useDialog,
+  UserStoreNodeFragment,
+} from '@openmsupply-client/common';
+import { RequestFragment, useRequest } from '../../api';
+import { useDraftRequisitionLine, useNextRequestLine } from './hooks';
+import { isRequestDisabled } from '../../../utils';
+import { RequestLineEdit } from './RequestLineEdit';
+import { Representation, RepresentationValue } from './utils';
+import { ItemWithStatsFragment } from '@openmsupply-client/system';
+
+interface RequestLineEditModalProps {
+  requisition: RequestFragment;
+  itemId: string | null;
+  isOpen: boolean;
+  onClose: () => void;
+  mode: ModalMode | null;
+  store?: UserStoreNodeFragment;
+}
+
+export const RequestLineEditModal = ({
+  requisition,
+  itemId,
+  isOpen,
+  onClose,
+  mode,
+  store,
+}: RequestLineEditModalProps) => {
+  const isDisabled = isRequestDisabled(requisition);
+  const { Modal } = useDialog({ onClose, isOpen });
+  const deleteLine = useRequest.line.deleteLine();
+
+  const lines = useMemo(
+    () =>
+      requisition?.lines.nodes
+        .slice()
+        .sort((a, b) => a.item.name.localeCompare(b.item.name)) ?? [],
+    [requisition?.lines.nodes]
+  );
+
+  const [currentItem, setCurrentItem] = useBufferState(
+    lines?.find(line => line.item.id === itemId)?.item
+  );
+  const [previousItemLineId, setPreviousItemLineId] = useBufferState<
+    string | null
+  >(null);
+  const [representation, setRepresentation] = useState<RepresentationValue>(
+    Representation.UNITS
+  );
+
+  const { draft, save, update } = useDraftRequisitionLine(currentItem);
+  const { hasNext, next } = useNextRequestLine(currentItem);
+
+  const isPacksEnabled = !!currentItem?.defaultPackSize;
+  const useConsumptionData =
+    store?.preferences?.useConsumptionAndStockFromCustomersForInternalOrders;
+  const isProgram = !!requisition?.program;
+  const nextDisabled = (!hasNext && mode === ModalMode.Update) || !currentItem;
+
+  const deletePreviousLine = () => {
+    if (previousItemLineId && !isDisabled) deleteLine(previousItemLineId);
+  };
+
+  const onCancel = () => {
+    if (mode === ModalMode.Create) {
+      deletePreviousLine();
+    }
+    onClose();
+  };
+
+  const onChangeItem = (item: ItemWithStatsFragment) => {
+    deletePreviousLine();
+    setRepresentation(Representation.UNITS);
+    setCurrentItem(item);
+  };
+
+  const onSave = async () => {
+    await save();
+    setPreviousItemLineId(null);
+    if (mode === ModalMode.Update && next) setCurrentItem(next);
+    else if (mode === ModalMode.Create) setCurrentItem(undefined);
+    else onClose();
+  };
+
+  useEffect(() => {
+    if (!!draft?.isCreated) {
+      save();
```

Comment:
Need to save once draft is created to get item stats on the created requisition line